include/zephyr/sys/util.h

```diff
@@ -546,6 +546,35 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 		(expr);                                                                            \
 	})
 
+/** @brief Calculate the base-2 logarithm */
+#define LOG2(x) (sizeof(uint64_t) * 8 - __builtin_clzll((uint64_t)(x)-1))
```

Comment:
You're super explicit with brackets and conditions elsewhere, its probably worth throwing brackets around `sizeof(uint64_t) * 8`

Reply:
RE the usage of generics in my original code, they do appear to be used in the Zephyr tree already in the cbprint internals: https://github.com/zephyrproject-rtos/zephyr/blob/9aa4cdbb67a45dcadb7c731de8d17bcb6679eedd/include/zephyr/sys/cbprintf_internal.h#L96
https://github.com/zephyrproject-rtos/zephyr/blob/9aa4cdbb67a45dcadb7c731de8d17bcb6679eedd/include/zephyr/sys/cbprintf_internal.h#L187
etc

Reply:
> You're super explicit with brackets and conditions elsewhere, its probably worth throwing brackets around `sizeof(uint64_t) * 8`

It's recommended to add brackets around any macro parameters (`x` in this case). Otherwise, it's just plain arithmetic.

> RE the usage of generics in my original code, they do appear to be used in the Zephyr tree already in the cbprint internals:

Yeah... maybe it is best to use generics here as well.